exemgr/main.cpp

```diff
@@ -1061,13 +1080,28 @@ class SessionThread
                     // here to make sure all syslogging from all the threads
                     // are complete; and that our logDbProfEndStatement will
                     // appear "last" in the syslog for this SQL statement.
-                    jl.reset();
-                    args.reset();
-                    args.add((int)csep.statementID());
-                    msgLog.logMessage(logging::LOG_TYPE_DEBUG,
-                                      logDbProfEndStatement,
-                                      args,
-                                      li);
+                    // puts the real destruction in another thread to avoid
+                    // making the whole session wait.  It can take several seconds.
+                    int stmtID = csep.statementID();
+                    std::unique_lock<std::mutex> scoped(jlMutex);
+                    // C7's compiler complains about the msgLog capture here
+                    // msgLog is global scope, and passed by copy, so, unclear
+                    // what the warning is about.
+                    destructing++;
+                    std::thread bgdtor([jl, &jlMutex, &jlCleanupDone, stmtID, &li, msgLog, &destructing] {
```

Comment:
My intellisense flags this msgLog parameter. 

> a variable with static storage duration cannot be captured in a lambda

Since it is static you can call from inside lambda without passing it into lambda in the first place.

Reply:
This isn't really an error, but after compiling, I noticed it causes an extra warning.

Reply:
good call.  I tried many different combinations of captures and capture types.  This one doesn't give a warning.  Will add.